src/Euclidean.jl

```diff
@@ -34,10 +36,11 @@ end
 det_local_metric(M::MetricManifold{<:Manifold,EuclideanMetric}, x) = one(eltype(x))
 
 inner(::Euclidean, x, v, w) = dot(v, w)
-inner(M::MetricManifold{<:Manifold,EuclideanMetric}, args...) = inner(manifold(M), args...)
+inner(::MetricManifold{<:Manifold,EuclideanMetric}, x, v, w) = dot(v, w)
```

Comment:
Wouldn't this require, that (for example for every matrix matrifold) `dot` does something resonable? I think it would either require the constructor for `MetricManifold{X,EuclideanMetric}` to avoid inconsistent constructions or this should default to an error.

I know what you are aiming at, I am just not yet sure how to define it without introducing too inconsistent ways of “misuse”, that might confuse the user then.

Reply:
Sure, but `dot` for matrix inputs is the Frobenius inner product, which to me seems the sensible choice when using a Euclidean metric. What would be a case where this default behavior wouldn't be sensible?

Reply:
Maybe I'd prefer not to use this default at all but rather use your neat `HasMetric` trait approach. My fear is (I think I wrote that), that you can equip a non-Euclidean metricised manifold (SPDs for example) and then get strange results here. Maybe I'd like to more avoid that than really criticising the choice of `dot`. And maybe this user error can also not be avoided here; I mean _if_ you equip SPDs oh Hyperbolic space with Euclidean – it'S your fault not the code. So yeah, maybe dot is the right choice.

Reply:
Yeah, I'm perfectly fine with letting the user make that error if they like, so long as the documentation and code is unambiguous about what `EuclideanMetric` means. There's actually some inconsistency here now. I'm working on a prototype interface for maps, which will let us construct pullback metrics. `Sphere` would then default to a pullback metric from an embedding map in R^{n+1} instead of just defaulting to the misleading `EuclideanMetric`.